datadog_checks_base/datadog_checks/base/checks/openmetrics/base_check.py

```diff
@@ -69,6 +79,8 @@ def __init__(self, *args, **kwargs):
             for instance in instances:
                 self.get_scraper_config(instance)
 
+        self.check_initializations.append(self.set_up_http_handlers)
```

Comment:
Why not have the `set_up_http_handlers` code directly here instead of `self.check_initializations`.

Is that because we want the error to be logged multiple time if there is an error?

Reply:
Yes